### PR TITLE
Fixed remote storage fallback.

### LINKED ISSUE #1253: Set compressor_file_storage to CompressorFileStorage
With the latest django-compressor release 4.5, the `compressor_file_storage` was set to `COMPRESS_STORAGE`:
https://github.com/django-compressor/django-compressor/commit/752edfaf0c4c10212792474479cca90d918c9d6a#diff-d2bf8e4739c94f5a9a6091b67159fc008af6e6ada59770261133cd3d854effc6R64

When using s3 as storage backend, the following error is returned:
`NotImplementedError: This backend doesn't support absolute paths.`

### Possible solution
```
compressor_file_storage = SimpleLazyObject(
    lambda: get_storage(alias="compressor_file_storage", storage_class="compressor.storage.CompressorFileStorage"
)
```

### PR DESCRIPTION
**Pull request by carltongibson**
Remote storages not implementing path need a fallback to a private instance of CompressorFileStorage. This must not be dependent on project settings.

Regression in v4.5. Closes #1253

### CODE REVIEW

File: compressor/storage.py [hunk 1]
```diff
@@ -10,10 +10,10 @@
...
 
 
-def get_storage(
-    alias=settings.COMPRESS_STORAGE_ALIAS,
-    storage_class=settings.COMPRESS_STORAGE
-):
+def get_storage(alias=None, storage_class=None):
+    alias = alias or settings.COMPRESS_STORAGE_ALIAS
+    storage_class = storage_class or settings.COMPRESS_STORAGE
+
```
Comment (carltongibson) [inline compressor/storage.py:16]: This is needed to allow override settings to work in the tests.

File: compressor/storage.py [hunk 2]
```diff
@@ -61,7 +61,9 @@ def save(self, filename, content):
 
 
 compressor_file_storage = SimpleLazyObject(
-    lambda: get_storage()
+    lambda: storages.create_storage(
+        {"BACKEND": "compressor.storage.CompressorFileStorage"}
+    )
```
Comment (carltongibson) [inline compressor/storage.py:66]: This is the actual fix. This **should not** be affected by user settings. (This was the error in v4.5)

File: compressor/tests/test_storages.py [hunk 3]
```diff
@@ -87,5 +101,35 @@ def test_offline_manifest_storage(self):
...
+        ),
+    )
+    def test_storage_without_path_fallback(self):
+        """
+        Remote storages not implementing path need a fallback to a private
+        instance of CompressorFileStorage. This must not be dependent on
+        project settings.
+        """
+        old_default_storage = storage.default_storage
+        storage.default_storage = storage.DefaultStorage()
```
Comment (carltongibson) [inline compressor/tests/test_storages.py:124]: get_filename uses the default storage, so we need to clear (and reset) in the test case when running more than just the single test.

**Review by saadmk11 — APPROVED** [review]
The code looks good, I haven't tested it though (on a vacation)

**Comment by dbartenstein** [comment]
@carltongibson looks good to me! Thanks! :bow:

**Comment by carltongibson** [comment]
Hey @diox — are you able to review. I think we could do a point release to fix the regression in 4.5 here. Thanks 🙏

**Review by diox — APPROVED** [review]

**Pull request merged**